### ISSUE TITLE
[Accepted with Revisions] SDL 0283 - Add Scrollable Message and Slider to Screen Manager

### ISSUE THREAD

**Issue by theresalech** [smartdevicelink/sdl_evolution#949]
Hello SDL community,

The review of "SDL 0283 - Add Scrollable Message and Slider to Screen Manager"  begins now and runs through March 17, 2020.  A previous review of this proposal took place February 26 - March 3, 2020.  The proposal is available here:

https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0283-screen-manager-scrollable-message-and-slider.md

Reviews are an important part of the SDL evolution process. All reviews should be sent to the associated Github issue at:

https://github.com/smartdevicelink/sdl_evolution/issues/949

What goes into a review?

The goal of the review process is to improve the proposal under review through constructive criticism and, eventually, determine the direction of SDL. When writing your review, here are some questions you might want to answer in your review:

* Is the problem being addressed significant enough to warrant a change to SDL?
* Does this proposal fit well with the feel and direction of SDL?
* If you have used competitors with a similar feature, how do you feel that this proposal compares to those?
* How much effort did you put into your review? A glance, a quick reading, or an in-depth study?
Please state explicitly whether you believe that the proposal should be accepted into SDL.

More information about the SDL evolution process is available at

https://github.com/smartdevicelink/sdl_evolution/blob/master/process.md

Thank you,
Theresa Lech

Program Manager - Livio
theresa@livio.io

**Comment by joeygrover** [comment]
While overall this proposal seems straightforward and helpful enough to accept, the proposal itself is hard to distinguish from the review in #948 of SDL 0282 Screen Manager Alert Manager. A lot of sections seem copy and paste, eg paragraph in Proposed solution and Potential Downside section. Is this proposal dependant on SDL 0282? If so, I don't think we should review this proposal until 0282 is reviewed and approved.

**Comment by joeljfischer** [comment]
I apologize, I didn't remember that I had made this proposal an appending of SDL-0282's alert manager. It is dependent on SDL-0282.

**Comment by theresalech** [comment]
Per the author's request, the Steering Committee voted to defer this proposal to allow the author to consider any impact the accepted revisions to [SDL 0282](#948) might have on this proposal and request changes as needed.

**Comment by theresalech** [comment]
It was determined that the accepted revisions to [SDL 0282](https://github.com/smartdevicelink/sdl_evolution/issues/948) did not require any changes to this proposal, so this proposal is now back in review.  The review will take place until 2020-03-17.

**Comment by joeljfischer** [comment]
I have noticed a few things in re-reading my proposal:

1\. 
```
/**
 This completion handler is used to return the user's selected position or an error if the slider fails. 
*/
typedef void(^SDLScreenManagerSliderCompletionHandler)(NSUInteger selectedPosition, NSError *__nullable error);
```

`selectedPosition` should be an `NSInteger` and the documentation should specify that the `selectedPosition` will be `-1` if there is an error.

2\. `extern NSTimeInterval SDLScreenManagerDefaultTimeout; // Is 0 in the implementation, used for View class timeouts to activate the `defaultTimeout`.` should be removed, it's not present in the Java Suite implementation and was an earlier idea that I decided not to use.

**Comment by theresalech** [comment]
The Steering Committee voted to accept this proposal with the revisions included in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/949#issuecomment-598229707) from the author.

**Comment by theresalech** [comment]
@joeljfischer please advise when a new PR has been entered to update the proposal to reflect the agreed upon revisions. I'll then merge the PR so the proposal is up to date, and enter issues in respective repositories for implementation. Thanks!

**Comment by theresalech** [comment]
Author has updated PR to reflect agreed upon revisions and implementation issues have been entered:
- [iOS](https://github.com/smartdevicelink/sdl_ios/issues/1597)
- [Java Suite](https://github.com/smartdevicelink/sdl_java_suite/issues/1317)
- [JavaScript Suite](https://github.com/smartdevicelink/sdl_javascript_suite/issues/141)
